### PR TITLE
Test cases: Base class for test cases, tests and fixes for utils.

### PR DESCRIPTION
**Pull request by cyberstar**
**Should be rebased** after merge of #11 and #12 since this PR depends on them.
Partially covers issue #4 

* Base class for test cases (some utility methods),
* Test cases to cover logic in utils.py (edge cases, normal cases),
* Fixes for some issues in utils .

Though test cases can be refactored to reduce boilerplate code.

### CODE REVIEW

File: aldryn_reversion/utils.py [hunk 1]
```diff
@@ -60,27 +60,25 @@ def get_translations_versions_for_object(obj, revision, versions=None):
...
+def get_fk_models(obj, blank=None):
     """
     Returns required (blank=False) FK models for given object.
     """
     fk_fields = [fk for fk in obj._meta.fields if type(fk) == ForeignKey]
-    required_fks_models = [fk.rel.to for fk in fk_fields if not fk.blank]
-    return required_fks_models
+    fks_models = []
+    for fk in fk_fields:
+        if (blank is not None and fk.blank == blank) or blank is None:
```
Comment (mkoistinen) [inline aldryn_reversion/utils.py]: Perhaps this would be simpler and easier to read?:

````
if blank is None or fk.blank == blank:
````

File: aldryn_reversion/utils.py [hunk 2]
```diff
@@ -102,16 +102,20 @@ def get_deleted_objects_versions(revision, versions=None,
 
 def get_conflict_fks_versions(obj, version, revision, exclude=None):
     """
-    Lookup for deleted FKs for object
+    Lookup for deleted FKs (blank=False)for obj, expects version to be obj
+    version from the same revision.
+    If exclude provided - excludes based on that from versions to check.
+    Expects exclude to be a dict of filter string, value i.e {'pk': 1}.
```
Comment (mkoistinen) [inline aldryn_reversion/utils.py]: BTW, thank you for using heredocs and comments! :trophy:

**Comment by mkoistinen** [comment]
Please consider changing the if statement above (check that its logically the same, could be you need a different order of evaluation, or there are side-effects to consider). Otherwise LGTM!

File: aldryn_reversion/utils.py (hunk 1 above)
Comment (cyberstar) [inline aldryn_reversion/utils.py]: You're right! Thanks for comment!

**Comment by cyberstar** [comment]
Rebased, also checked and changed the if statement.
